src/controllers/tripRequestsSearchController.js

```diff
@@ -0,0 +1,224 @@
+/* eslint-disable import/prefer-default-export */
+import * as tripRequest from '../services/searchTripRequest';
+import * as notFound from '../utils/errors/notFoundError';
+import * as rolesHelper from '../utils/helpers/userRole';
+import * as applicationError from '../utils/errors/applicationsErrors';
+
+const tripRequestSearch = async (req, res) => {
+  if (req.query.departure && req.query.destination) {
+    try {
+      const { departure, destination } = req.query;
+      const role = await rolesHelper.getUserRole(req.user.id);
+      const finds = await tripRequest.searchByDepartureDestination(
+        departure,
+        destination,
+        req.user.id,
+        role
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
+      return notFound.isNotFound(
+        { data: { message: 'Not found, try again!' } },
+        res
+      );
+    } catch (error) {
+      return applicationError.internalServerError(
+        { data: { message: `Error: ${error} ,try again!` } },
+        res
+      );
+    }
+  }
+  if (req.query.startDate && req.query.endDate) {
+    try {
+      const { startDate, endDate } = req.query;
+      const role = await rolesHelper.getUserRole(req.user.id);
+      const finds = await tripRequest.searchByStartEndDate(
+        startDate,
+        endDate,
+        req.user.id,
+        role
+      );
+      if (finds) {
+        return res.status(200).json({ status: 200, data: { trips: finds } });
+      }
+      return notFound.isNotFound(
+        { data: { message: 'Not found, try again!' } },
+        res
+      );
+    } catch (error) {
+      return applicationError.internalServerError(
+        { data: { message: `Error: ${error} ,try again!` } },
+        res
+      );
+    }
+  }
+  if (req.query.requestId) {
```

Comment:
some one will not probably search with an ID. so use a full-text search here. You should be able to search all trips with a certain name, eg the name of an accommodation or a place, or a description